core/indexer/processTransactions.go

```diff
@@ -120,20 +120,30 @@ func (tdp *txDatabaseProcessor) prepareTransactionsForDatabase(
 
 func (tdp *txDatabaseProcessor) addScResultInfoInTx(scr *smartContractResult.SmartContractResult, tx *Transaction) *Transaction {
 	dbScResult := tdp.commonProcessor.convertScResultInDatabaseScr(scr)
-	if tx.Sender != dbScResult.Receiver || len(dbScResult.Data) == 0 {
-		return tx
-	}
+	// now all smart contract results that belongs to a transaction are indexed
+	//if tx.Sender != dbScResult.Receiver || len(dbScResult.Data) == 0 {
+	//	return tx
+	//}
 
 	tx.SmartContractResults = append(tx.SmartContractResults, dbScResult)
 
-	if dbScResult.GasLimit != 0 && dbScResult.Value != "0" {
+	if isSCWithGasUsed(dbScResult, tx) {
 		gasUsed := tx.GasLimit - scr.GasLimit
 		tx.GasUsed = gasUsed
 	}
 
 	return tx
 }
 
+func isSCWithGasUsed(dbScResult ScResult, tx *Transaction) bool {
```

Comment:
no need for this function

Reply:
check if prevTxHash == originalTxHash == txHash

Reply:
done.